docker/create_cluster.py

```diff
@@ -1,67 +1,62 @@
-    user_defined_networks = [network for network in networks if network.name not in predefined_networks]
-
+    # If a network exists with the name of the one you requested to create
+    # remove the network and create a new one
+    user_defined_networks = [network for network in networks if network.name == network_name]
     for user_defined_network in user_defined_networks:
-        print('Removing network: {}'.format(user_defined_network.name))
-        user_defined_network.remove()
-
-    # Verify that all user defined networks are removed
-    networks = docker_client.networks.list()
-    user_defined_networks = [network for network in networks if network.name not in predefined_networks]
-
-    if len(user_defined_networks) != 0:
-        raise DockerError('Failed to remove all networks!')
 
+        print('Removing containers for network ({})'.format(user_defined_network.name))
 
-def remove_containers(docker_client):
-    """ Stops / removes all containers """
+        # Get docker containers for the network
+        network_containers = user_defined_network.containers
+        for network_container in network_containers:
 
-    containers = docker_client.containers.list(all=True)
-    # TODO: filter by network name prefix, so it only removes containers in current network
-    for container in containers:
+            # Remove the containers attached to this network
+            print('Removing container: {}'.format(network_container.name))
+            subprocess.check_call(['docker', 'stop', network_container.name])
+            subprocess.check_call(['docker', 'rm', '-f', network_container.name])
 
-        # HACK: Calling stop and remove via docker-py will timeout frequently
-        print('Stopping / removing container: {}'.format(container.name))
-        subprocess.check_call(['docker', 'stop', container.name])
-        subprocess.check_call(['docker', 'rm', container.name])
+        # Remove the network
+        print('Removing network {}'.format(user_defined_network.name))
+        user_defined_network.remove()
 
-    # Verify that all containers have been removed
-    containers = docker_client.containers.list(all=True)
-    if len(containers) != 0:
-        raise DockerError('Failed to remove all containers!')
+    # Verify that all user defined networks are removed
+    networks = docker_client.networks.list()
+    user_defined_network = [network for network in networks if network.name == network_name]
+    if len(user_defined_network) != 0:
+        raise DockerError('Failed to remove all networks!')
 
 
-def create_cluster(clean, network_name, number_of_nodes, public_key_path):
+def create_cluster(pull, clean, network_name, number_of_nodes, public_key_path):
 
     docker_client = docker.from_env()
 
-    print('Pulling sethrosetter/centos7-systemd-sshd image ...')
-    docker_client.images.pull('sethrosetter/centos7-systemd-sshd')
+    if pull:
+        print('Pulling sethrosetter/centos7-systemd-sshd image ...')
```

Comment:
I would like to propose that this docker clean up not block this PR as I will be working on it right after in its own branch and there will be a lot of changes.